specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/common/2022-03-01/parameters.json

```diff
@@ -0,0 +1,54 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2022-03-01",
+    "title": "Common types and definitions",
+    "description": "Definitions and parameters common for all resource types of KubernetesConfiguration RP"
+  },
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {},
+  "parameters": {
+    "ClusterRpParameter": {
+      "name": "clusterRp",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The Kubernetes cluster RP - i.e. Microsoft.ContainerService, Microsoft.Kubernetes, Microsoft.HybridContainerService.",
+      "x-ms-parameter-location": "method"
```

Comment:
In our previous release, this parameter is defined by an enumeration, please consider to introduce that back, this brings unnecessary breaking changes to the SDKs.
See previous definitions here: https://github.com/Azure/azure-rest-api-specs/blob/a11e5d03047c44a2a506079292f5b80e4df45fb4/specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/common/2022-01-01-preview/parameters.json#L34

Reply:
The reason why we are making this breaking change is because we are extending the set of parent resource providers and parent resource types that we support for our extension resource. These will increase over time and we don't want to release a new api-version/Swagger when we want to extend the parent provider set so this was changed to string

Reply:
Make sense